### ISSUE TITLE
post - working with git

### ISSUE THREAD

**Issue by sergueyarellano** [sergueyarellano/sergueyarellano.github.io#1]
You work in a repo in `github`, you have friends or teammates and want to be more efficient with your development. Here is how we do it.

# Branches

- `master` - reflects the code that is in production.
- `develop` - it is created from `master`. `feat/*` branches will integrate into `develop`.
- `feat/name` - are created from `develop`. `feat/account-form`, `feat/button-animation`, etc... they are work in progress (WIP), and when complete, they integrate into `develop`.
- `fix/name` - are created from `develop`. They reflect changes that solve a bug in the codebase and will integrate into `develop`

```
        H---I---J feat/button-animation
      /
    D---E---F---G develop
  /
A---B---C master
```

# Create feature branch

Checkout branch develop, update and checkout a new feature branch
```
$ git checkout develop
$ git pull
$ git checkout -b feat/button-animation
```

If you set an `alias`
```
$ git config --global alias.cd !sh -c "git checkout develop; git pull"
```

next time you can do
```
$ git cd
$ git checkout -b feat/button-animation
```

# Workflow in a future branch

One rule, if you rebase (that is what you should do IMO) you should not share a feature branch with another person. [Checkout the perils of rebasing](https://git-scm.com/book/en/v2/Git-Branching-Rebasing)

The reason is that rebasing a branch `re-hashes` the new commits that you have in the **feature** branch, integrates new commits from the branch you are rebasing, and finally puts that re-hashed commits **on top** of the **feature** branch. 

Basically, you are changing the `base/parent` commit from where the feature branch was created to the latest commit that came in.

before rebasing
```
    H---I---J feat/button-animation
  /
D---E---F---G develop
```

After rebasing:

```
                H'---I'---J' feat/button-animation
              /
D---E---F---G develop
```
